### ISSUE TITLE
Signal App crashes when tapping on settings / "linked devices"

### ISSUE THREAD

**Issue by MavisTodd** [signalapp/Signal-Android#12338]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I'm trying to link a Signal Desktop instance with my Signal Android app. Every time I open "Settings" in the android app and tap on "linked devices" I get the error / crash-message "Signal angehalten / Bericht / OK" [probably: Signal unexpectedly stopped / Report / OK]. Then it returns to the start / message page. The other settings-options work alright, signal in general is a bit slow and sometimes crashes, but in no way as consistently. I already cleared cache, cleared google play services cache, restarted phone, checked if version is latest, deleted all data and restored from backup, all makes no change.
It is (probably) rather likely that this bug is due to my very old phone/android version (also because i found no other hits for the same problem). but nonetheless it is strange, because signal otherwise works (more or less) fine.

### Steps to reproduce
- start signal
- go to settings
- tap on "linked devices"

**Actual result:** 
Signal crashes and does not open "linked devices"-menu. instead after some time it returns to message view or has to be restarted altogether.
**Expected result:** 
Linked devices menu should be shown so that i be able to scan a QR code and link a device.

### Device info
**Device:** Samsung GT-i9195 (don't laugh)
**Android version:** 4.4.2 (dito)
**Signal version:** 5.42.7

### Link to debug log
https://debuglogs.org/android/5.42.7/cf78b418b9290b9fd8b75f39d9dbe0a6b7474e1352973832bf3314f1989299c5

I am really not very competent with this android stuff, or rather: not at all. So explanations for dummies would be appreciated.
Thank you in advance.
M.

**Comment by alex-signal** [comment]
Hi!

This'll be fixed in 5.43, which should be going to beta next week.

TL;DR - Android 4.4 doesn't support vectors natively, you need to use Android's compatibility library, which we didn't in this case.

If you're curious why this happened, we recently made a bunch of changes to update the user interface (UI) to Google's new Material3 design system. That included changing the + button at the bottom of the linked devices screen to use a different component to display it. When updating that, I inadvertently used the wrong thing to set the button's image.

Android 4.4 doesn't support vectors out of the box, but the support libraries (or "compat" libraries) do. But this means we need to use their special stuff to set some things (like vector images on different ui elements).

We use vectors because they'll render correctly and without artifacting (aliasing for example) across all devices and they're much smaller than shipping lots and lots of webp or png files.

**Comment by alex-signal** [comment]
(REopening because this'll auto-close once the beta goes out)

**Comment by MavisTodd** [comment]
Thank you, Alex, for the answer. I didn't understand the details, but got that I simply have to wait for the new version. (As for the + button - i even don't get to the point where that is visible, but perhaps you meant that trying to load this causes signal to crash.)
Anyway, thanks!

**Comment by fulldecent** [comment]
I am experiencing this crash also: 

1. Get a iPhone (13)
2. turn on lockdown mode
3. Install Signal, open it
4. say no on popups for: notifications, local network, contacts
5. Get a mac (M1 MBP, 2021)
6. turn on lockdown mode
7. Install Signal
8. Open Signal, get QR code
9. Scan QR code from camera app on iPhone
10. It opens Signal
11. Signal says it needs a camera to link a device (even though you already deep linked into Signal)
12. say no on popup for camera
13. go to settings app to turn on camera
14. scan qr code again, it opens signal app
15. signal says it wants to scan qr code
16. click to scan qr code from inside signal app
17. say no to popup for microphone

Expected outcome:

1. Camera access is not required for Signal to link a device with deep linking
2. If you do want to allow camera access, linking a device from inside the app should link the device

Actual outcome:

1. Deep links do not work
2. Unnecessary permissions are requested many times inside the app
3. Crashes and linking devices is not possible
19. iOS app crashes after step 17

**Comment by greyson-signal** [comment]
@fulldecent Thanks for the report, but this is where we track bugs for the android app. You can open up an issue for the iOS app over here: https://github.com/signalapp/Signal-iOS

**Comment by fulldecent** [comment]
Sorry, thank you @greyson-signal  Reported to https://github.com/signalapp/Signal-iOS/issues/5705
